ouroboros-consensus-test/bench-mempool/Main.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedLists    #-}
+
+
+module Main (main) where
+
+import           Control.Monad (void)
+import           Data.Set ()
+import           Test.Tasty.HUnit (testCase, (@?=))
+
+import qualified Control.Tracer as Tracer
+import           Test.Tasty.Bench (bench, bgroup, defaultMain, nfIO)
+
+import           Bench.Consensus.Mempool
+import           Bench.Consensus.Mempool.TestBlock (TestBlock)
+import qualified Bench.Consensus.Mempool.TestBlock as TestBlock
+import           Bench.Consensus.MempoolWithMockedLedgerItf
+import qualified Ouroboros.Consensus.Mempool.Capacity as Mempool
+
+main :: IO ()
+main = defaultMain [ benchmarkJustAddingTransactions ]
+  where
+    benchmarkJustAddingTransactions =
+        bgroup "Just adding transactions"
+          $  fmap benchAddNTxs [1_000, 10_000, 100_000, 1_000_000]
+          <>
+             [ testCase "Transactions are added" $ do
+                 let
+                   n   = 1000
+                   txs = mkNTryAddTxs 1000
+                 mempool    <- addNTxs txs
+                 mempoolTxs <- getTxs mempool
+                 mempoolTxs @?= txsAddedInCmds txs
+                 length mempoolTxs @?= n
+             ]
```

Comment:
To be sure, I would run this test for each of the `n` you use in the benchmarks above here, e.g., by moving the test into `benchAddNTxs`. Each benchmark then has a sanity check

Another addition would be to add a regular quickcheck test that run the test for arbitrary values of `n`, such that we're also leveraging property testing in addition to sanity checks for the benchmarks

Reply:
Good point. In this case, I wanted to avoid the cost of having to test each of the values of `n`. Eventually, we should test the interpretation of commands by means of the mempool property tests, but I think it makes sense to test each case.